### PR TITLE
fix quniform issue ; change related doc

### PR DESCRIPTION
**Pull request by suiguoxin**
#1377

### CODE REVIEW

File: docs/en_US/Tutorial/SearchSpaceSpec.md [hunk 1]
```diff
@@ -45,7 +45,8 @@ All types of sampling strategies and their parameter are listed here:
   * When optimizing, this variable is constrained to a two-sided interval.
 
 * {"_type":"quniform","_value":[low, high, q]}
-  * Which means the variable value is a value like round(uniform(low, high) / q) * q
+  * Which means the variable value is a value like randint(np.floor((high-low)/q)+1)*q + low. For example, for _value specified as [0, 10, 2.5], possible values are [0, 2.5, 5.0, 7.5, 10.0]; For _value specified as [2, 10, 5], possible values are [2, 7]. All possible values have the same possibility to be selected.
```
Comment (xuehui1991) [inline docs/en_US/Tutorial/SearchSpaceSpec.md]: no need to add "low" here.

File: docs/en_US/Tutorial/SearchSpaceSpec.md [hunk 2]
```diff
@@ -45,15 +45,16 @@ All types of sampling strategies and their parameter are listed here:
   * When optimizing, this variable is constrained to a two-sided interval.
 
 * {"_type":"quniform","_value":[low, high, q]}
-  * Which means the variable value is a value like round(uniform(low, high) / q) * q
+  * Which means the variable value is a value like randint(floor((high-low)/q)+1)*q + low. For example, for _value specified as [0, 10, 2.5], possible values are [0, 2.5, 5.0, 7.5, 10.0]; For _value specified as [2, 10, 5], possible values are [2, 7]. All possible values have the same possibility to be selected.
```
Comment (xuehui1991) [inline docs/en_US/Tutorial/SearchSpaceSpec.md]: Also need to update here https://github.com/microsoft/nni/blob/master/docs/en_US/Tutorial/AnnotationSpec.md

File: src/sdk/pynni/nni/parameter_expressions.py [hunk 3]
```diff
@@ -57,7 +57,7 @@ def quniform(low, high, q, random_state):
     q: sample step
     random_state: an object of numpy.random.RandomState
```
Comment (xuehui1991) [inline src/sdk/pynni/nni/parameter_expressions.py]: need to change here:
https://github.com/microsoft/nni/blob/master/src/sdk/pynni/nni/smartparam.py

File: docs/en_US/Tutorial/SearchSpaceSpec.md (hunk 2 above)
Comment (suiguoxin) [reply docs/en_US/Tutorial/SearchSpaceSpec.md]: updated

File: src/sdk/pynni/nni/parameter_expressions.py (hunk 3 above)
Comment (suiguoxin) [reply src/sdk/pynni/nni/parameter_expressions.py]: updated

File: docs/en_US/Tutorial/AnnotationSpec.md [hunk 4]
```diff
@@ -41,11 +41,11 @@ There are 10 types to express your search space as follows:
 * `@nni.variable(nni.uniform(low, high),name=variable)`
   Which means the variable value is a value uniformly between low and high.
 * `@nni.variable(nni.quniform(low, high, q),name=variable)`
-  Which means the variable value is a value like round(uniform(low, high) / q) * q
+  Which means the variable value is a value like randint(floor((high-low)/q)+1)*q + low
 * `@nni.variable(nni.loguniform(low, high),name=variable)`
   Which means the variable value is a value drawn according to exp(uniform(low, high)) so that the logarithm of the return value is uniformly distributed.
 * `@nni.variable(nni.qloguniform(low, high, q),name=variable)`
-  Which means the variable value is a value like round(exp(uniform(low, high)) / q) * q
+  Which means the variable value is a value like floor((loguniform(low, high) - low) / q) * q + low
```
Comment (QuanluZhang) [inline docs/en_US/Tutorial/AnnotationSpec.md]: please reconsider this logic

Comment (suiguoxin) [reply docs/en_US/Tutorial/AnnotationSpec.md]: not clear, what's the risk

**Comment by QuanluZhang** [comment]
please check all the tuners and advisors to see whether these changes take effect, as some tuners/advisors may not use the changed function.

**Comment by suiguoxin** [comment]
> please check all the tuners and advisors to see whether these changes take effect, as some tuners/advisors may not use the changed function.
Quite different among different tuners, i.e.: 
Hyperband/GPTuner/evolution_tuner: parameter_expression used
BOHB: Package Configspace used,  no difference between qloguniform/ quniform

**Pull request merged**